Main.py

```diff
@@ -2,431 +2,368 @@
 
 import discord
 from discord.ext import commands
-import asyncio
-import sqlite3
-from tabulate import tabulate
-import logging
-import os, sys
+
 import datetime
+import logging
 import math
+import os
 import pytz
+import sqlite3
+import statistics
 import subprocess
+import sys
 
-DB_PATH = './Scoreboard.db'
+CMD_PREFIX = "%"
+CROSSWORD_ROLE = "crosswords"
+CROSSWORD_TIMEZONE = "America/New_York"
+DB_PATH = "./Scoreboard.db"
+DEVELOPER_ROLE = "idoneam"
 
-cmd_prefix = '%'
-bot = commands.Bot(command_prefix=cmd_prefix)
+bot = commands.Bot(command_prefix=CMD_PREFIX)
 
 # Logging configuration
 logger = logging.getLogger('discord')
 logger.setLevel(logging.ERROR)
-handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode = 'w')
+handler = logging.FileHandler(filename='discord.log', encoding='utf-8', mode='w')
 handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
 logger.addHandler(handler)
 
+
+def _format_time(time) -> str:
+    """
+    Formats total seconds into a time with minutes and padded seconds if more than 59 seconds.
+    """
+    if time > 59:
+        m = int(math.floor(time / 60))
+        s = int(time % 60)
+        return f"{m}:{str(s).zfill(2)}"
+    return str(int(time))
+
+
+def _from_ymd(time_str: str) -> datetime.datetime:
+    return datetime.datetime.strptime(time_str, "%Y-%m-%d")
+
+
+def _as_ymd(dt: datetime.datetime) -> str:
+    return dt.strftime("%Y-%m-%d")
+
+
+def _get_day(dt: datetime.datetime) -> str:
+    return dt.strftime("%a")
+
+
+def _get_day_from_ymd(time_str) -> str:
+    return _get_day(_from_ymd(time_str))
+
+
 @bot.event
-@asyncio.coroutine
-def on_ready():
-    print('Logged in as {0} ({1})'.format(bot.user.name, bot.user.id))
-
-@bot.command(pass_context=True)
-@commands.has_role("idoneam")
-@asyncio.coroutine
-async def update(self):
+async def on_ready():
+    print(f'Logged in as {bot.user.name} ({bot.user.id})')
```

Comment:
I would also put this in the log

Reply:
19e11bd1e01522040608edc461f68e8d7edf8cc6